diff -ur xchat-gnome-0.4-o/src/fe-gnome/navigation-tree.c xchat-gnome-0.4/src/fe-gnome/navigation-tree.c
--- xchat-gnome-0.4-o/src/fe-gnome/navigation-tree.c	2005-05-13 18:15:29.000000000 -0600
+++ xchat-gnome-0.4/src/fe-gnome/navigation-tree.c	2005-06-11 10:23:06.000000000 -0600
@@ -1227,15 +1227,17 @@
 navigation_model_path_deref (NavModel * model, GtkTreePath * path)
 {
     gint ref_count;
+		gboolean worked;
     GtkTreeIter iter;
     GtkTreePath *unsorted = gtk_tree_model_sort_convert_path_to_child_path (GTK_TREE_MODEL_SORT (model->sorted),
 									    path);
 
-    gtk_tree_model_get_iter (GTK_TREE_MODEL (model->store), &iter, unsorted);
-    gtk_tree_model_get (GTK_TREE_MODEL (model->store), &iter, 5, &ref_count, -1);
+		if(gtk_tree_model_get_iter (GTK_TREE_MODEL (model->store), &iter, unsorted)) {
+			gtk_tree_model_get (GTK_TREE_MODEL (model->store), &iter, 5, &ref_count, -1);
 
-    if (ref_count > 0)
-	gtk_tree_store_set (model->store, &iter, 5, ref_count - 1, -1);
+			if (ref_count > 0)
+				gtk_tree_store_set (model->store, &iter, 5, ref_count - 1, -1);
+	}
 
     gtk_tree_path_free (unsorted);
 }
