diff -ur xchat-gnome-0.7-0/src/fe-gnome/navigation-tree.c xchat-gnome-0.7/src/fe-gnome/navigation-tree.c
--- xchat-gnome-0.7-0/src/fe-gnome/navigation-tree.c	2005-11-13 23:50:10.000000000 -0700
+++ xchat-gnome-0.7/src/fe-gnome/navigation-tree.c	2005-12-17 19:00:21.000000000 -0700
@@ -1053,7 +1053,7 @@
 
 	treeview = GTK_TREE_VIEW (glade_xml_get_widget (gui.xml, "userlist"));
 
-	if (gui.server_tree->current_path != NULL)
+	if (gui.server_tree && gui.server_tree->current_path != NULL)
 		navigation_model_path_deref (gui.server_tree->model, gui.server_tree->current_path);
 
 	/* XXX: This sets model to be the GtkTreeModelSort used by the NavTree, it is
