diff -ur xchat-gnome-0.16-o/data/icons/hicolor/128x128/apps/Makefile.am xchat-gnome-0.16/data/icons/hicolor/128x128/apps/Makefile.am
--- xchat-gnome-0.16-o/data/icons/hicolor/128x128/apps/Makefile.am	2006-12-10 15:27:44.000000000 -0700
+++ xchat-gnome-0.16/data/icons/hicolor/128x128/apps/Makefile.am	2007-01-10 09:16:42.000000000 -0700
@@ -1,4 +1,4 @@
-themedir = $(pkgdatadir)/icons/hicolor
+themedir = $(datadir)/icons/hicolor
 category = apps
 size = 128x128
 
diff -ur xchat-gnome-0.16-o/data/icons/hicolor/48x48/apps/Makefile.am xchat-gnome-0.16/data/icons/hicolor/48x48/apps/Makefile.am
--- xchat-gnome-0.16-o/data/icons/hicolor/48x48/apps/Makefile.am	2006-12-10 15:27:43.000000000 -0700
+++ xchat-gnome-0.16/data/icons/hicolor/48x48/apps/Makefile.am	2007-01-10 09:16:42.000000000 -0700
@@ -1,4 +1,4 @@
-themedir = $(pkgdatadir)/icons/hicolor
+themedir = $(datadir)/icons/hicolor
 category = apps
 size = 48x48
 
